### ISSUE TITLE
Push notifications not enabled (while in-app message works)

### ISSUE THREAD

**Issue by ithav** [intercom/intercom-cordova#65]
Please use the following template to submit your issue. Following this template will allow us to quickly investigate and help you with your issue. Please be aware that issues which do not conform to this template may be closed.

For feature requests please contact us at team@intercom.io

## Version info
  - Intercom for Cordova/PhoneGap version: phonegap 6.3 / intercom 1.1.7 (**note**: I'm using the highest plugin version that passed the phonegap build without errors)
  - iOS/Android OS version: 5.1.1
  - iOS/Android Device: Nexus 4

## Expected behavior
while installing the app with the intercom plugin, the push permission would appear and needs to be approved. later on, push notifications sent from Intercom dashboard should get to the device/app.

## Actual behavior
push notifications permission is not part of the required set of permission during installation
once installed, in-app message are getting to the app successfully, but looking at Intercom dashboard it seems that push notifications are not enabled.

## Steps to reproduce
 1. upload to build.phonegap.com a zipped source with config.xml file containing those settings:
<preference name="intercom-app-id" value="*******"/>
<preference name="intercom-android-api-key" value="android_sdk-**************************"/>
<preference name="intercom-android-sender-id" value="**********"/>
<gap:plugin name="cordova-plugin-intercom" source="npm" version="1.1.7" /> 
 2. trigger a DEBUG build, download and install the APK file
 3. 

## Logs

**note**: I only followed the first 2 steps at https://docs.intercom.io/configure-intercom-for-your-product-or-site/configure-intercom-for-mobile/enable-push-notifications-with-intercom-for-android.
since my app is not a native android app & I'm using phonegap cloud build service.

Any help or guidance would be appreciated.

**Comment by bboyle18** [comment]
Hi ithav!
Thanks for reporting this issue.
I was wondering if you could tell me what build errors, if any, you were seeing?

Thanks
Brian

**Comment by ithav** [comment]
Hi Brian,

meanwhile I was able to overcome the original issue.
I think that this part: https://github.com/intercom/intercom-cordova#push-notifications
should be modified from:
```
To enable iOS push notifications, simply call intercom.registerForPush().
To enable Android push notifications, your Sender ID to your app's config.xml as follows:
<preference name="intercom-android-sender-id" value="YOUR_SENDER_ID"/>
```
into:
`To enable push notifications (for both iOS or Android), simply call intercom.registerForPush(YOUR_SENDER_ID, successCallback, errorCallback).
in addition, To enable Android push notifications, your Sender ID to your app's config.xml as follows:
<preference name="intercom-android-sender-id" value="YOUR_SENDER_ID"/>`

I was mislead by the current text.

now, regarding your request, this is the error I'm getting when trying to build with version 3.0.3:

> * What went wrong:
> A problem occurred configuring root project 'project'.
> > Could not resolve all dependencies for configuration ':_debugCompile'.
>    > Could not find com.google.android.gms:play-services-gcm:9.2.1.
>      Searched in the following locations:
>          https://repo1.maven.org/maven2/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.pom
>          https://repo1.maven.org/maven2/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.jar
>          https://jcenter.bintray.com/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.pom
>          https://jcenter.bintray.com/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.jar
>          https://dl.bintray.com/intercom/intercom-maven/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.pom
>          https://dl.bintray.com/intercom/intercom-maven/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.jar
>          file:/android-sdk/extras/android/m2repository/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.pom
>          file:/android-sdk/extras/android/m2repository/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.jar
>          file:/android-sdk/extras/google/m2repository/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.pom
>          file:/android-sdk/extras/google/m2repository/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.jar
>      Required by:
>          :project:unspecified > io.intercom.android:intercom-sdk:3.0.2 > io.intercom.android:intercom-sdk-gcm:3.0.2
> 

Thanks for your time!
Ithav

**Comment by jtreanor** [comment]
Thanks @ithav,

Can you try building your app on PhoneGap Build using version [3.0.4](https://github.com/intercom/intercom-cordova/releases/tag/3.0.4)? This version improves compatibility for iOS across different Cordova versions.

Unfortunately the Android build error you posted is not an Intercom error. This issue occurs when the machine building the app does not have an up to date "Google Repository" installed. I'm not familiar with PhoneGap Build so I'm unsure how this should be resolved.

**Comment by ithav** [comment]
Hello.
I tried to build with 3.0.**5** but it failed for Android with:
`* What went wrong:
A problem occurred configuring root project 'project'.
> Could not resolve all dependencies for configuration ':_debugCompile'.
   > Could not find com.google.android.gms:play-services-gcm:9.2.1.
     Searched in the following locations:
         https://repo1.maven.org/maven2/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.pom
         https://repo1.maven.org/maven2/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.jar
         https://jcenter.bintray.com/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.pom
         https://jcenter.bintray.com/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.jar
         https://dl.bintray.com/intercom/intercom-maven/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.pom
         https://dl.bintray.com/intercom/intercom-maven/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.jar
         file:/android-sdk/extras/android/m2repository/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.pom
         file:/android-sdk/extras/android/m2repository/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.jar
         file:/android-sdk/extras/google/m2repository/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.pom
         file:/android-sdk/extras/google/m2repository/com/google/android/gms/play-services-gcm/9.2.1/play-services-gcm-9.2.1.jar
     Required by:
         :project:unspecified > io.intercom.android:intercom-sdk:3.0.2 > io.intercom.android:intercom-sdk-gcm:3.0.2`

and failed for iOS with:
`Ld build/MyAppQA.build/Release-iphoneos/MyAppQA.build/Objects-normal/arm64/MyAppQA normal arm64
    cd /project
    export IPHONEOS_DEPLOYMENT_TARGET=8.0
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS9.1.sdk -L/project/build -L/project/MyAppQA/Plugins/cordova-plugin-intercom -F/project/build -filelist /project/build/MyAppQA.build/Release-iphoneos/MyAppQA.build/Objects-normal/arm64/MyAppQA.LinkFileList -miphoneos-version-min=8.0 -dead_strip -ObjC -fobjc-arc -fobjc-link-runtime /project/build/libCordova.a /project/MyAppQA/Plugins/cordova-plugin-intercom/Intercom -framework Foundation -framework UIKit -framework Accelerate -framework Security -framework SystemConfiguration -framework ImageIO -framework QuartzCore -framework AudioToolbox -framework CFNetwork -licucore -lxml2 -framework CoreTelephony -Xlinker -dependency_info -Xlinker /project/build/MyAppQA.build/Release-iphoneos/MyAppQA.build/Objects-normal/arm64/MyAppQA_dependency_info.dat -o /project/build/MyAppQA.build/Release-iphoneos/MyAppQA.build/Objects-normal/arm64/MyAppQA
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_PHImageRequestOptions", referenced from:
      objc-class-ref in Intercom(ITBUpload.o)
  "_OBJC_CLASS_$_PHPhotoLibrary", referenced from:
      objc-class-ref in Intercom(ITBUpload.o)
  "_OBJC_CLASS_$_PHImageManager", referenced from:
      objc-class-ref in Intercom(ITBUpload.o)
  "_OBJC_CLASS_$_PHAsset", referenced from:
      objc-class-ref in Intercom(ITBUpload.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **


The following build commands failed:
	Ld build/MyAppQA.build/Release-iphoneos/MyAppQA.build/Objects-normal/arm64/MyAppQA normal arm64
(1 failure)`

**Comment by bboyle18** [comment]
Hi @ithav 👋

This issue is going to be fixed in our next release. 
In the meantime, if you want to include the fix yourself you can add 
`<framework src="Photos.framework" />`
to your plugin.xml. The exact change is detailed in this [Pull Request ](https://github.com/intercom/intercom-cordova/pull/68/files)

Please let me know if this fixes your issue.
Cheers
Brian

**Comment by jtreanor** [comment]
@ithav This error is now fixed in [3.0.6](https://github.com/intercom/intercom-cordova/releases/tag/3.0.6).

**Comment by ithav** [comment]
Hi @jtreanor 

with 3.0.6 the phonegap build passes succesfully (iOS and Android).
It did force me to restrict the iOS version to >8.0, but I can live with that.

Thanks a lot for you support!
Ithav
